### PR TITLE
Plugin plot subcomponent

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request refactors the existing bqplot instances inside plugins to use a new re-usable `Plot` "plugin subcomponent" which both simplifies the code in the plugins themselves and adds support for popping out any plot into its own window.

This currently updates the implementation for existing plots in Imviz's aperture photometry and line profiles, but is not expected to result in any changes to the user except for the new addition of the popout button (plots should otherwise look and act the same as before this PR).

image

In the future, we can extend on this subcomponent to add support for viewer tools (pan, zoom, etc)

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/core/template_mixin.py [hunk 1]
```diff
@@ -2484,3 +2486,92 @@ def export_table(self):
...
+    Plot subcomponent mixin.
+
+    In addition to ``plot``, this provides the following methods at the plugin-level:
+
+    * :meth:`clear_plot`
+
+    To render in the plugin's vue file::
+
+      <jupyter-widget :widget="plot_widget"></jupyter-widget>
+    """
```
Comment (pllim) [inline jdaviz/core/template_mixin.py:2565]:
```suggestion
    """  # noqa: E501
```

**Review by pllim — APPROVED** [review]
Able to pop out the plot does help a lot when there are too much info to display or when the axis label all get squished together in original GUI.

But, the popout does not follow dark mode. Is this a deal breaker? 😉 

Screenshot 2023-06-14 154601

**Review by javerbukh — APPROVED** [review]
Looks good! Ditto about dark mode though 😄

**Comment by kecnry** [comment]
none of the popouts obey dark mode... if that's something we want, it might be possible to pass along that information to the window somehow, but I'm not sure.

**Comment by mariobuikhuizen** [comment]
@pllim @javerbukh I can take a look at getting dark mode in the popouts

**Comment by pllim** [comment]
Thanks, Mario! I opened follow up issue at https://github.com/spacetelescope/jdaviz/issues/2270

**Pull request merged**